src/ast.ts

```diff
@@ -144,10 +160,26 @@ export function reference(base: Exp, prop: string | Exp): ExpReference {
 // immutability of the original expression).
 export function copyExp(exp: Exp): Exp {
   exp = <Exp> util.extend({}, exp);
-  if (exp.type === 'op' || exp.type === 'call') {
-    (<ExpOp> exp).args = util.copyArray((<ExpOp> exp).args);
+  switch (exp.type) {
+  case 'op':
```

Comment:
'op' and 'call' have the same structure - they both need to copy their args (array) property.